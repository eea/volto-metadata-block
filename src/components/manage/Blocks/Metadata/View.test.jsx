import React from 'react';
import { Provider } from 'react-intl-redux';
import { render, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import ViewMetadataBlock from './View';
import '@testing-library/jest-dom/extend-expect';

const mockStore = configureStore();

const store = mockStore({
  content: {
    data: {
      test: 'Test Output',
      nonexistent: 'Placeholder Output',
    },
  },
  intl: {
    locale: 'en',
    messages: {},
  },
});

const mockData = {
  data: {
    id: 'test',
    placeholder: 'Placeholder Output',
  },
  properties: {
    test: 'Property Output',
  },
  metadata: {
    test: 'Metadata Output',
  },
};

jest.mock('@plone/volto/registry', () => ({
  widgets: {
    views: {
      getWidget:
        (data) =>
        ({ value, className }) => <div className={className}>{value}</div>,
      default: ({ value, className }) => (
        <div className={className}>{value}</div>
      ),
    },
  },
}));

describe('ViewMetadataBlock', () => {
  it('renders correctly with widget output', () => {
    render(
      <Provider store={store}>
        <ViewMetadataBlock data={mockData} />
      </Provider>,
    );
    expect(screen.getByText('Test Output')).toBeInTheDocument();
  });

  it('renders placeholder if no output is available', () => {
    const placeholderData = {
      data: {
        id: 'nonexistent',
        placeholder: 'Placeholder Output',
      },
    };

    render(
      <Provider store={store}>
        <ViewMetadataBlock data={placeholderData} />
      </Provider>,
    );

    expect(screen.getByText('Placeholder Output')).toBeInTheDocument();
  });

  it('does not render if data id is not present', () => {
    const emptyData = {
      data: {},
    };

    const { container } = render(
      <Provider store={store}>
        <ViewMetadataBlock data={emptyData} />
      </Provider>,
    );

    expect(container.innerHTML).toBe('');
  });
});
