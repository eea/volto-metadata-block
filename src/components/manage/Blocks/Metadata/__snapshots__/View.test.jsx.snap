// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewMetadataBlock renders correctly 1`] = `
<div>
  Mock Widget
</div>
`;
